framework/player/init/groupTracker.sqf

```diff
@@ -0,0 +1,128 @@
+if !(isNil "Grp_SWD4") then {[Grp_SWD4,"b_mech_inf","SWORD4",1] call _lmf_fnc_createIcon};
+
+if !(isNil "Grp_HMR1") then {[Grp_HMR1,"b_armor","HAMMER1",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_HMR2") then {[Grp_HMR2,"b_armor","HAMMER2",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_HMR3") then {[Grp_HMR3,"b_armor","HAMMER3",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_HMR4") then {[Grp_HMR4,"b_armor","HAMMER4",1] call _lmf_fnc_createIcon};
+
+//AIR
+if !(isNil "Grp_FLC1") then {[Grp_FLC1,"b_air","FALCON1",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_FLC2") then {[Grp_FLC2,"b_air","FALCON2",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_FLC3") then {[Grp_FLC3,"b_air","FALCON3",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_FLC4") then {[Grp_FLC4,"b_air","FALCON4",1] call _lmf_fnc_createIcon};
+
+if !(isNil "Grp_CDR1") then {[Grp_CDR1,"b_air","CONDOR1",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_CDR2") then {[Grp_CDR2,"b_air","CONDOR2",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_CDR3") then {[Grp_CDR3,"b_air","CONDOR3",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_CDR4") then {[Grp_CDR4,"b_air","CONDOR4",1] call _lmf_fnc_createIcon};
+
+if !(isNil "Grp_HWK1") then {[Grp_HWK1,"b_air","HAWK1",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_HWK2") then {[Grp_HWK2,"b_air","HAWK2",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_HWK3") then {[Grp_HWK3,"b_air","HAWK3",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_HWK4") then {[Grp_HWK4,"b_air","HAWK4",1] call _lmf_fnc_createIcon};
+
+if !(isNil "Grp_EGL1") then {[Grp_EGL1,"b_plane","EAGLE1",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_EGL2") then {[Grp_EGL2,"b_plane","EAGLE1",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_EGL3") then {[Grp_EGL3,"b_plane","EAGLE1",1] call _lmf_fnc_createIcon};
+if !(isNil "Grp_EGL4") then {[Grp_EGL4,"b_plane","EAGLE1",1] call _lmf_fnc_createIcon};
+
+
+//EVEN HANDLERS ///////////////////////////////////////////////////////////////////////////////////
+//WAIT UNTIL TIME IS BIGGER THAN 1 (to avoid intro glitches)
+waitUntil {time > 1};
+
+//MOUSE HOVER OVER ICON EH
+private _hoverEH = addMissionEventHandler ["GroupIconOverEnter",{
+	params ["_is3D","_group"];
+
+	//ICON PARAMETERS
+	_group setGroupIconParams [var_iconColor,groupID _group,1.3,true];
+}];
+
+//MOUSE HOVER LEAVE ICON EH
+private _hoverLeaveEH = addMissionEventHandler ["GroupIconOverLeave",{
+	params ["_is3D","_group"];
+
+	//ICON PARAMETERS
+	_group setGroupIconParams [var_iconColor,groupID _group,1,true];
+
+	//FADE TEXT
+	"lmf_layer2" cutFadeOut 0;
+}];
+
+//ON CLICKING ICON
+private _clickIconEH = addMissionEventHandler ["GroupIconClick",{
+	params ["_is3D","_group","_waypointId","_mouseButton"];
+
+	//CHECK FOR LEFT CLICK
+	if (_mouseButton isEqualTo 0) then {
+		//GET ARRAY WITH UNIT NAMES FROM GROUP
+		private _nameArray = [];
+		{
+			private _name = name _x;
+			_nameArray pushBackUnique _name;
```

Comment:
Doubt we have two John Does, so `pushbackunique` is probably not needed. Also no need for assigning a variable if you do not use the name a second time.